### DISCUSSION TITLE
"Saved Search" Option / Dynamic Collections

### DISCUSSION THREAD

**Discussion by ozornayashoujo** [ppy/osu#21022] [Ideas] [answered]
Often times players want to play a specific set of maps, be it small CS, high OD, maps by one or several mappers etc. This implies typing out a long search query every time. One solution is players making collections. They pick and choose maps they think fit a certain skill set and play them in quick succession to practice one or several of many aspects of osu!

My suggestion is to make it possible to "Save" a search query for future use. I.e. you type out: "sotarks star>=5 star<6 bpm<200 ar<9.5" for a bunch of farm maps. Perhaps next to the search bar a small icon suggesting to "Save as Collection", which could be in the list of all collection with a small lightning symbol (⚡) implying the collection is dynamic and not hand-made.

In my opinion this would improve player experience, removing the frustration of losing collections or having to make custom ones yourself if you just want some specific parameters without the hassle of "Right-click > Add to Collection > *scroll* > Click" dozens of times. And remember - that's just one collection. If you're dedicated to learning some particular BPM range you don't want to search for it every time typing out the query or have to resort to "Sort by BPM", since it ignores star rating or other sorting method which the player might prefer.

I simply think this would help people quickly find what they want and even discover new maps they might have downloaded but never played. Simply put - this would increase involvement and play time.

**Comment by ozornayashoujo** [comment]
P.S.
Such an option would also effectively enable sorting by criteria like CS, OD, AR to some extent. I think that cluttering the Sorting tab with every single parameter is a bad idea, so why not let those who want something like that create dynamic collections? You can already search the osu! website by map parameters but once you download the map itself it gets mixed in with everything else. I think it makes sense at all levels of play, and adding that on top of the star rating limiter would allow people to cherry pick maps more efficiently.

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu/discussions/13010
